magenta-lib/src/main/scala/magenta/tasks/ChangeSetTasks.scala

```diff
@@ -0,0 +1,150 @@
+   stack: Stack,
+   createStackIfAbsent:Boolean,
+   alwaysUploadToS3:Boolean,
+   changeSetName: String
+)(implicit val keyRing: KeyRing, artifactClient: AmazonS3) extends Task {
+
+  override def execute(reporter: DeployReporter, stopFlag: => Boolean) = if (!stopFlag) {
+    val cfnClient = CloudFormation.makeCfnClient(keyRing, region)
+    val s3Client = S3.makeS3client(keyRing, region)
+    val stsClient = STS.makeSTSclient(keyRing, region)
+    val accountNumber = STS.getAccountNumber(stsClient)
+
+    val maybeCfStack = cloudFormationStackLookupStrategy match {
+      case LookupByName(cloudFormationStackName) => CloudFormation.describeStack(cloudFormationStackName, cfnClient)
+      case LookupByTags(tags) => CloudFormation.findStackByTags(tags, reporter, cfnClient)
+    }
+
+    val templateString = templatePath.fetchContentAsString.right.getOrElse(
+      reporter.fail(s"Unable to locate cloudformation template s3://${templatePath.bucket}/${templatePath.key}")
+    )
+
+    val stackName = UpdateCloudFormationTask.nameToCallNewStack(cloudFormationStackLookupStrategy)
+    val template = processTemplate(stackName, templateString, s3Client, stsClient, region, alwaysUploadToS3, reporter)
+
+    val templateParameters = CloudFormation.validateTemplate(template, cfnClient).getParameters.asScala
+      .map(tp => TemplateParameter(tp.getParameterKey, Option(tp.getDefaultValue).isDefined))
+
+    val resolvedAmiParameters: Map[String, String] = amiParameterMap.flatMap { case (name, tags) =>
+      val ami = latestImage(accountNumber)(region.name)(tags)
+      ami.map(name ->)
+    }
+
+    val parameters: Map[String, ParameterValue] =
+      combineParameters(stack, stage, templateParameters, userParameters ++ resolvedAmiParameters)
+
+    reporter.info("Creating Cloudformation change set")
+    reporter.info(s"Stack name: $stackName")
+    reporter.info(s"Change set name: $changeSetName")
+    reporter.info(s"Parameters: $parameters")
+
+    val stackTags = PartialFunction.condOpt(cloudFormationStackLookupStrategy){ case LookupByTags(tags) => tags }
+    val changeSetType = if(maybeCfStack.isEmpty) { ChangeSetType.CREATE } else { ChangeSetType.UPDATE }
+
+    if(changeSetType == ChangeSetType.CREATE && !createStackIfAbsent) {
+      reporter.fail(s"Stack $cloudFormationStackLookupStrategy doesn't exist and createStackIfAbsent is false")
+    } else {
+      CloudFormation.createChangeSet(reporter, changeSetName, changeSetType, stackName, stackTags, template, parameters, cfnClient)
+    }
+  }
+
+  def description = s"Create change set $changeSetName for stack $cloudFormationStackLookupStrategy with ${templatePath.fileName}"
+  def verbose = description
+}
+
+case class CheckChangeSetCreatedTask(
+  region: Region,
+  cloudFormationStackLookupStrategy: CloudFormationStackLookupStrategy,
+  changeSetName: String,
+  duration: Long
+)(implicit val keyRing: KeyRing, artifactClient: AmazonS3) extends Task with RepeatedPollingCheck {
```

Comment:
Not for this PR, but it might be worth looking into the AWS SDK "Waiter" functionality to replace `RepeatedPollingCheck`.

https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/cloudformation/waiters/AmazonCloudFormationWaiters.html